src/figure/formation.c

```diff
@@ -186,10 +189,84 @@ void formation_record_fight(formation *m)
     m->recent_fight = 6;
 }
 
+int is_formation_halted(const formation *m)
+{
+    for (int i = 0; i < m->num_figures; i++) {
+        int figure_id = m->figures[i];
+        if (figure_id) {
+            const figure *f = figure_get(figure_id);
+            if (f->direction != DIR_8_NONE) {
+                return 0; // At least one figure is moving
+            }
+        }
+    }
+    return 1; // All figures are idle or invalid
+}
+
+int is_formation_moving(formation *m)
```

Comment:
```suggestion
static int is_formation_moving(formation *m)
```

Also, this function has side effects and doesn't just calculate whether a formation is moving.

Please either change the function name to be more explicit or separate the side effects to a different function.